### DISCUSSION TITLE
Partial Downloads

### DISCUSSION THREAD

**Discussion by pikabelle** [aclap-dev/video-downloadhelper#1494] [Help and Q&A]
### Your question

Hi. Ever since I bought the videodownloadhelper for $28, some of my videos are partially downloading. I tried to search for this in the forum, but cannot find it. Can I not contact the developers since I bought it? Before I bought it, it never partially downloaded anything. I'm not closing my browser or tab out. Please help! I'm very frustrated since I spent hard earned money on it. Thanks!

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"08b7a863c4b94cdf"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:132.0) Gecko/20100101 Firefox/132.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "C:\\Users\\pikab\\Briefcase\\Download Helper",
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "yt_warning": false,
    "open_count_store": 6,
    "successfull_dl": 69
}

**Comment by Mattmjs** [comment]
It's probably a coincidence that you think buying a license is causing issues for your downloads. If you provide an example video that
it happens on then it can be investigated.

**Reply by pikabelle** [reply]
Thank you for your reply. Here is an example video: https://www.youtube.com/watch?v=pPDBn9pSLE4
Thanks!

**Reply by Mattmjs** [reply]
No problem, it downloaded the whole video.

**Comment by pikabelle** [comment]
It didn’t download all of it on mine. And I tried several times. What do you think might be going on with mine? I had 2 other videos I downloaded on Sunday that got cut off too. Here are their links:
https://www.youtube.com/watch?v=2fkBMbTxFYE
https://www.youtube.com/watch?v=BatwLNk06eI

**Reply by Mattmjs** [reply]
VPN ? YouTube premium? They can cause problems. This might not matter but what resolution do 
you choose ?

**Comment by pikabelle** [comment]
I don't have any of those. I uninstalled and reinstalled Downloadhelper and put the key back in, but that didn't help. Two new videos I downloaded were cut off. I uninstalled and reinstalled Downloadhelper and didn't put the key back in and the videos downloaded fully, so it must have something to do with the key. I'm very disappointed. :(

**Reply by Mattmjs** [reply]
Your first example worked on windows 10 and the other 2 I tried on Linux which also worked 100%. No partial download.
I can't explain what's happening on your system, could it be your network connection is momentarily disconnecting.
It would show that happening on the task bar if that was the case. Firefox just got updated so maybe try with the new version.

**Reply by Mattmjs** [reply]
I forgot to point out that I don't have a license 
on windows while on Linux a license is not 
needed. In other words no water mark added
to video just as it would behave as those who
have the license on windows and Mac.

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I don't have an answer for you.  Like Matt, I don't generally experience any issues downloading from YouTube.  I have some random thoughts.

- Does there seem to be a pattern in the file size or duration of what gets downloaded?  I'm wondering if these files always cut off at a particular size or duration.  Or does it seem to be at random sizes & durations?
- It might be educational & informative to monitor your downloads in the Resource Monitor & the network status display.  These are both tools built into Windows.  You can launch Resource Monitor in a command prompt like this:

perfmon /res

You'll want to focus on the Network tab.  Here's what mine looks like at the moment, but I'm not downloading anything right now.

![#01](https://github.com/user-attachments/assets/de4f55df-bfeb-49b7-af41-8ad77ea3c1ec)

You can get the network status display like this:

powershell "explorer ""::{208D2C60-3AEA-1069-A2D7-08002B30309D}\::{7007ACC7-3202-11D1-AAD2-00805FC1270E}\::$((Get-NetAdapter -Name Ethernet*).DeviceID)"""

That's a single command on a single line.  It's folded here by GitHub to make it fit.  This command generates something that looks like this:

![#02](https://github.com/user-attachments/assets/52bdb2d1-7dc0-44ae-a42a-08f1ab743d55)

There is a way to find this through the standard system Settings dialog but I never remember how to navigate to it.  So I use this command (in a .bat file, of course) instead.  In the bizarre command, the token `Ethernet` is a name that can be different on each system.  On my system, when I execute this command . . .

![#03](https://github.com/user-attachments/assets/8b2836b9-8da6-4fc8-b5f7-6adfbfd08dfa)

This is the beginning of the results of the command on my system.  There's a bunch more output from the command but that's irrelevant here.  The relevant bit is the name on the line that starts with `Ethernet adapter`.  The last token on that line is the name of your internet connection.  Use that name in the bizarre Power Shell command.

I didn't figure that command out.  It's too bizarre.  I found it in a discussion on elevenforum.  Great resource.  It's helped me a lot, as did sevenforums before it.  (Those aren't typos.  One is singular & the other is plural.)  I believe there is also a tenforum but I have never run Windows 10 so you're on your own with that.

In any case, my idea is to carefully monitor what goes on in those things during your downloads.  Don't stare at the VDH status display.  Stare at these things.  I don't know what you'd be looking for.  But maybe something will show up that will mean something to us.
